File: leibniz.js
let pi = 4;
let sign = -1;
let start_at = Date.now();
for (let i = 3; i < 2147483647; i += 2) {
    pi = pi + sign * 4 / i;
    sign *= -1;
}
console.log(pi + " consume " + (Date.now() - start_at) + " msec");

File: leibniz.go
package main

import (
	"fmt"
	"time"
)

func main() {
	var (
		pi       float32 = 4.0
		sign     float32 = -1.0
		start_at         = time.Now().UnixMilli()
	)
	for i := 3; i < 2147483647; i += 2 {
		pi = pi + (sign * 4.0 / float32(i))
		sign *= -1
	}
	fmt.Println(pi, " consume ", (time.Now().UnixMilli() - start_at), " msec")
}
